No time library for freeBSD?

We just recently tried to update from `v1.7.0` to `v1.47.7`(I know, huge jump), and it looks like in that jump the time functions were split out to OS specific version (linux, mac, and windows). Unfortunately, this breaks our [freebsd/amd64 build](https://github.com/grafana/loki/actions/runs/10744853958/job/29802951861) complaining of:

```
vendor/github.com/fsouza/fake-gcs-server/internal/backend/fs.go:120:82: undefined: createTimeFromFileInfo
vendor/github.com/fsouza/fake-gcs-server/internal/backend/fs.go:156:82: undefined: createTimeFromFileInfo
```

Is this a known issue? Should it/can it use the linux version? Is support for freeBSD intentionally being dropped? Please let me know if I can help so we can upgrade our very out of date version. Thanks!

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -16,7 +16,7 @@ jobs:
         go_version:
           - "1.22"
           - "1.23"
-        os:
+        goos:
           - freebsd
           - macos
           - ubuntu

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -9,6 +9,36 @@ on:
       - main
 
 jobs:
+  # Validates that we can compile on the targets below
+  compile:
+    strategy:
+      matrix:
+        go_version:
+          - "1.22"
+          - "1.23"
+        os:
+          - freebsd
+          - macos
+          - ubuntu
+          - windows
+        goarch:
+          - amd64
+          - arm64
+
+    name: cross-compilation (GOOS=${{ matrix.goos }}, GOARCH=${{ matrix.goarch }})
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.1.7
+
+      - uses: actions/setup-go@v5.0.2
+        with:
+          go-version: "1.x"
+
+      - run: go build
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+
   test:
     strategy:
       matrix:

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -17,9 +17,9 @@ jobs:
           - "1.22"
           - "1.23"
         goos:
+          - darwin
           - freebsd
-          - macos
-          - ubuntu
+          - linux
           - windows
         goarch:
           - amd64
@@ -49,14 +49,8 @@ jobs:
           - macos
           - ubuntu
           - windows
-        arch:
-          - 386
-          - amd64
-        exclude:
-          - os: macos
-            arch: 386
 
-    name: tests (${{ matrix.os }}/go-${{ matrix.go_version }}-${{ matrix.arch }})
+    name: tests (${{ matrix.os }}/go-${{ matrix.go_version }}
     runs-on: ${{ matrix.os }}-latest
     steps:
       - uses: actions/checkout@v4.1.7
@@ -67,16 +61,7 @@ jobs:
           go-version: ${{ matrix.go_version }}
 
       - name: run-tests-race
-        if: ${{ matrix.arch == 'amd64' }}
-        env:
-          GOARCH: "${{ matrix.arch }}"
         run: go test -race -vet all -mod readonly ./...
-
-      - name: run-tests
-        if: ${{ matrix.arch == '386' }}
-        env:
-          GOARCH: "${{ matrix.arch }}"
-        run: go test -vet all -mod readonly ./...
 
   lint:
     name: lint

--- a/internal/backend/time_bsd.go
+++ b/internal/backend/time_bsd.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build darwin || freebsd
+
 package backend
 
 import (

